src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -27,6 +55,15 @@ public OssSecretMasker(IEnumerable<RegexPattern> patterns)
     private OssSecretMasker(OssSecretMasker copy)
     {
         _secretMasker = copy._secretMasker.Clone();
+
+        if (copy.TelemetryEnabled)
+        {
+            _charsScanned = copy._charsScanned;
```

Comment:
consider encapsulating these fields into a nested class and managing that class-level object.  that would enable managing the state a bit more atomically (e.g. _detectionTelemetry and _detectionAction can be readonly and initialized in the nested class constructor)

Reply:
Good idea! Done.